src/backend/fs.rs

```diff
@@ -3,151 +3,127 @@
 use std::{
     fs::{self, OpenOptions},
     io::{self, Read, Write},
-    //path::Path,
     path::{Path, PathBuf},
 };
 
-use anyhow::{anyhow, Result};
-use carbonado::{
-    constants::{Format, FEC_M},
-    fs::Header,
-    structs::Encoded,
-};
+use anyhow::{anyhow, Error, Result};
+use axum::body::Bytes;
+use carbonado::{constants::Format, file::Header, structs::Encoded};
+use futures_util::{stream::BoxStream, StreamExt, TryStreamExt};
 use log::{debug, trace};
 use rayon::prelude::*;
 use secp256k1::{ecdh::SharedSecret, PublicKey, SecretKey};
 
 use crate::{config::SYS_CFG, prelude::*};
 
-pub async fn write_file(pk: &Secp256k1PubKey, file_bytes: &[u8]) -> Result<Blake3Hash> {
-    debug!("Write file, len: {}", file_bytes.len());
+pub type FileStream<'a> = BoxStream<'a, Result<Bytes>>;
 
-    trace!("Create a shared secret using ECDH");
+pub async fn write_file<'a>(
+    pk: &Secp256k1PubKey,
+    file_stream: FileStream<'a>,
+) -> Result<Blake3Hash> {
+    trace!("write_file, create a shared secret using ECDH");
     let sk = SYS_CFG.private_key;
-    let ss = SharedSecret::new(&pk.into_inner(), &sk);
-    let write_pk = PublicKey::from_secret_key_global(&SecretKey::from_slice(&ss.secret_bytes())?);
+    let ss = SharedSecret::new(&pk.into_inner(), &sk).secret_bytes();
```

Comment:
Does ".secret_bytes()" refer to length or size?
as
secp256k1::ecdh::SharedSecret
pub fn secret_bytes(&self) -> [u8; SHARED_SECRET_SIZE]
I am not clear how it effects, write_pk